src/NuGetForUnity/Editor/NugetPackageVersion.cs

```diff
@@ -0,0 +1,602 @@
+                        {
+                            PreRelease = version.Substring(preReleaseStartIndex + 1);
+
+                            version = version.Substring(0, preReleaseStartIndex);
+                        }
+
+                        var split = version.Split('.');
+                        major = int.Parse(split[0]);
+                        minor = 0;
+                        if (split.Length >= 2)
+                        {
+                            minor = int.Parse(split[1]);
+                        }
+
+                        patch = 0;
+                        if (split.Length >= 3)
+                        {
+                            patch = int.Parse(split[2]);
+                        }
+
+                        revision = 0;
+                        if (split.Length >= 4)
+                        {
+                            revision = int.Parse(split[3]);
+                        }
+
+                        return;
+                    }
+                    catch (Exception ex)
+                    {
+                        Debug.LogErrorFormat("Invalid version number: '{0}'\n{1}", version, ex);
+                    }
+                }
+
+                buildMetadata = null;
+                PreRelease = null;
+                major = -1;
+                minor = -1;
+                patch = -1;
+                revision = -1;
+            }
+
+            public string PreRelease { get; }
+
+            /// <summary>
+            ///     Compares two version numbers in the form "1.2". Also supports an optional 3rd and 4th number as well as a prerelease tag, such as
+            ///     "1.3.0.1-alpha2".
+            ///     Returns:
+            ///     -1 if this is less than other
+            ///     0 if this is equal to other
+            ///     +1 if this is greater than other.
+            /// </summary>
+            /// <param name="other">The second version number to compare.</param>
+            /// <returns>-1 if this is less than other. 0 if this is equal to other. +1 if this is greater than other.</returns>
+            public int Compare(in SemVer2Version other)
+            {
+                try
+                {
+                    var majorComparison = major.CompareTo(other.major);
+                    if (majorComparison == 0)
+                    {
+                        // if major versions are equal, compare minor versions
+                        var minorComparison = minor.CompareTo(other.minor);
+                        if (minorComparison == 0)
+                        {
+                            var patchNumber = patch;
+                            var otherPatch = other.patch;
+                            var patchComparison = patchNumber.CompareTo(otherPatch);
+                            if (patchComparison == 0)
+                            {
+                                // if patch versions are equal, compare build versions
```

Comment:
```suggestion
                                // if patch versions are equal, compare revision versions
```
We might want to change comments as wording has changed and it might be confusing.